### DISCUSSION TITLE
Is there a straightforward way to downgrade packages?

### DISCUSSION THREAD

**Discussion by JodiWarren** [changesets/changesets#788] [Q&A]
Somehow a developer (on one of his last days) managed to incorrectly push a major version bump through for a package, which has led to a cavalcade of major bumps throughout our monorepo.

Is there a straightforward way of downgrading the versions of these packages? It's for an internal-only repo so we're not too concerned with breaking external dependencies.

**Comment by Andarist** [comment]
Hmm, this is tricky - mainly because you can't simply "remove" a version once it got published. And it depends on how many packages you have there etc. I would say:
- either roll with it - if it's internal, why does it matter?
- revert the versioning commit, fix up version bumps and do a fresh release. However, once some of those packages actually reach a new major... it won't be possible to publish it over an already published version so you will have a problem there in the future

What is your ideal scenario here? I wonder how we could make this kind of stuff easier.

**Reply by JodiWarren** [reply]
Our solution for this was to find and replace a lot of version numbers across the repo, and then do a new set of releases with patch bumps. So something like:

```
v1.1.0
|
v2.0 <- incorrect
|
v1.1.1 <- manual change
|
v1.1.2 <- changeset bump to update dependencies
```

Ideally it wouldn't have mattered, but it's a big legacy codebase where we aren't able to be as nimble as we may have liked.

I suppose my hope was that changeset would have a feature where `changeset add` would also expose the option of `downgrade`, which would then update the dependency tree. It would probably need variations of patch, minor, and major. It should probably also be behind a flag, as you definitely wouldn't want anyone selecting that normally. 

However, it was easy enough to work around that, so maybe that should be the preferred method instead of increasing the featureset.
